tutorial/10_key_features/005_visualization.py

```diff
@@ -28,109 +26,49 @@
-torch.manual_seed(SEED)
 
-###################################################################################################
-# Before defining the objective function, prepare some utility functions for training.
-def train_model(model, optimizer, train_loader):
-    model.train()
-
-    for batch_idx, (data, target) in enumerate(train_loader):
-        if batch_idx * BATCH_SIZE >= N_TRAIN_SAMPLES:
-            break
-        data, target = data.view(data.size(0), -1).to(DEVICE), target.to(DEVICE)
-
-        optimizer.zero_grad()
-        output = model(data)
-        loss = F.nll_loss(output, target)
-        loss.backward()
-        optimizer.step()
-
-
-def eval_model(model, valid_loader):
-    model.eval()
-    correct = 0
-    with torch.no_grad():
-        for batch_idx, (data, target) in enumerate(valid_loader):
-            if batch_idx * BATCH_SIZE >= N_VALID_SAMPLES:
-                break
-            data, target = data.view(data.size(0), -1).to(DEVICE), target.to(DEVICE)
-            output = model(data)
-            pred = output.argmax(dim=1, keepdim=True)
-            correct += pred.eq(target.view_as(pred)).sum().item()
-    accuracy = correct / min(len(valid_loader.dataset), N_VALID_SAMPLES)
-    return accuracy
+np.random.seed(SEED)
 
 
 ###################################################################################################
 # Define the objective function.
 def objective(trial):
-    train_loader = torch.utils.data.DataLoader(
-        torchvision.datasets.MNIST(
-            DIR, train=True, download=True, transform=torchvision.transforms.ToTensor()
-        ),
-        batch_size=BATCH_SIZE,
-        shuffle=True,
-    )
-    valid_loader = torch.utils.data.DataLoader(
-        torchvision.datasets.MNIST(
-            DIR, train=False, download=True, transform=torchvision.transforms.ToTensor()
-        ),
-        batch_size=BATCH_SIZE,
-        shuffle=True,
+    data, target = sklearn.datasets.load_breast_cancer(return_X_y=True)
+    train_x, valid_x, train_y, valid_y = train_test_split(data, target, test_size=0.25)
+    dtrain = lgb.Dataset(train_x, label=train_y)
+    dvalid = lgb.Dataset(valid_x, label=valid_y)
+
+    param = {
+        "objective": "binary",
+        "metric": "auc",
+        "verbosity": -1,
+        "boosting_type": "gbdt",
+        "lambda_l1": trial.suggest_float("lambda_l1", 1e-8, 10.0, log=True),
+        "lambda_l2": trial.suggest_float("lambda_l2", 1e-8, 10.0, log=True),
+        "num_leaves": trial.suggest_int("num_leaves", 2, 256),
+        "feature_fraction": trial.suggest_float("feature_fraction", 0.4, 1.0),
+        "bagging_fraction": trial.suggest_float("bagging_fraction", 0.4, 1.0),
+        "bagging_freq": trial.suggest_int("bagging_freq", 1, 7),
+        "min_child_samples": trial.suggest_int("min_child_samples", 5, 100),
+    }
+
+    # Add a callback for pruning.
+    pruning_callback = optuna.integration.LightGBMPruningCallback(trial, "auc")
+    gbm = lgb.train(
```

Comment:
Seems like it takes 10 minutes to run 4 trials. So it hasn't gotten particularly faster. It is because of the pruning? https://circleci.com/api/v1.1/project/github/optuna/optuna/64524/output/104/0?file=true&allocation-id=5fcf45d6dabe8b01b85d5353-0-build%2F69E17DC0 Can we reduce the number of boosting rounds or somehow shorten the time for each trial?